src/sql/src/plan/statement/ddl.rs

```diff
@@ -5012,13 +5091,56 @@ pub fn describe_alter_object_swap(
 }
 
 pub fn plan_alter_object_swap(
-    _: &mut StatementContext,
-    _: AlterObjectSwapStatement,
+    scx: &mut StatementContext,
+    stmt: AlterObjectSwapStatement,
 ) -> Result<Plan, PlanError> {
-    Err(PlanError::Unsupported {
-        feature: "ALTER ... SWAP ...".to_string(),
-        issue_no: Some(12972),
-    })
+    let AlterObjectSwapStatement {
+        object_type,
+        name_a,
+        name_b,
+    } = stmt;
+    let object_type = object_type.into();
+
+    // We'll try 10 times to generate a temporary suffix.
+    let gen_temp_suffix = |check_fn: &dyn Fn(&str) -> bool| {
+        let mut attempts = 0;
+        let name_temp = loop {
+            attempts += 1;
+            if attempts > 10 {
+                tracing::warn!("Unable to generate temp id for swapping");
+                sql_bail!("unable to swap!");
+            }
+
+            // To make these temporary names a bit more manageable, we make them short, by using
```

Comment:
The uuid v4 [spec](https://www.rfc-editor.org/rfc/rfc4122#section-4.4) says that bits 6, 7, 12-15 are all hard coded, so using the first 8 characters is less random than we might expect. Probably doesn't matter because there's enough other random bits in the first 8 chars though?

Reply:
Wow great call!! Changed this to use the last characters of a v4 UUID instead which is 12 instead of 8 and gives us better randomness because it doesn't use any of the hard coded bits